### DISCUSSION TITLE
Consider moving some feature crates into the main repo (like `slog-term` and `slog-async`)

### DISCUSSION THREAD

**Discussion by Techcable** [slog-rs/slog#302] [Ideas]
Right now, there are a number of seperate feature crates in the main `slog-rs` organization:

Many of these do not have their own issue tracker, and they often seem to fall behind the main crate 😦 

It might make sense to either:
1. Make them part of the `workspace` in the main repository
    - They could still be *separate dependencies*, but updating would be done in lockstep (and we'd have a unified issue tracker)
2. Keep them as separate repos, but enable github issues.

I would note that some of these crates are effectively official (and very stable). Keeping them as separate repositories doesn't really add much to the main crate 😉 

### Things to merge
Crates I propose integrating into the main repo, as part of the main workspace:
1. `slog-term`
2. `slog-json`
2. `slog-async`
3. `slog-stdlog`

These crates are essential to most `slog` users and the ecosystem as a whole. It makes sense to provide the same level of support as we do in the main repo.

Disadvantages:
1. We would have to offer the same level of support for these crates
2. We would have to make a message in the old repo saying things have moved

Advantages:
1. If a change in `slog` breaks one of these crates, we will know right away
   - Just run `cargo check --workspace` and we will get a message (and GH actions will fail)
2. Unified bug tracker, so users don't get confused 
   - Right now we have a lot of users getting bugs with `slog-term`, then filing it on the main repo.

Honestly, I think it's worth it so that we can provide first-class support for these crates :)

Note that *user-level dependencies would still be separate*. Adding `slog` as a dependency wouldn't automatically add `slog-term` or `slog-json`.
### Things to keep seperate
Crates I propose keeping separate (for now):
1. `slog-scope`
2. `slog-journald` 
3. `slog-syslog`
4. `slog-bunyan` 

For all of these, we should enable github issues, so people have a proper place to put bugs 😉 

I think (1) should be separate (probably permanently) because it's discouraged.

Crates 2-4 seem like more niche logging formats. Maybe we could move them in eventually........

**Comment by dpc** [comment]
Fine with me.

**Comment by Techcable** [comment]
Honestly I don't think this change is worth it. The overall setup seems to work fine right now.

However, it might be nice to enable separate issue trackers 😉
